platform/linux-generic/odp_event.c

```diff
@@ -24,17 +25,24 @@ odp_event_type_t odp_event_type(odp_event_t event)
 
 odp_event_subtype_t odp_event_subtype(odp_event_t event)
 {
-	return _odp_buffer_event_subtype(odp_buffer_from_event(event));
+	if (_odp_buffer_event_type(odp_buffer_from_event(event)) !=
+			ODP_EVENT_PACKET)
+		return ODP_EVENT_NO_SUBTYPE;
+
+	return packet_subtype(odp_packet_from_event(event));
 }
```

Comment:
Looking on this code. Maybe we can change this part so that subtypes do exist only for packets from the API point of view? So instead of `odp_event_subtype()` we will have `odp_packet_subtype()`.

Reply:
That's the situation today, but in the future we may add subtypes for also other event types. With odp_event_subtype() the application has to do only a single API call compared to e.g. odp_packet_subtype() where an application would have to always call odp_event_type() first.

Reply:
ok, fine